### ISSUE TITLE
math.intersect throws error for non-intersecting lines when all parameters are matrices

### ISSUE THREAD

**Issue by trent2** [josdejong/mathjs#2269]
```javascript
var a = math.matrix([1,0,0]);
var b = math.matrix([1,1,1])
var c = math.matrix([0,1,0])
var d = math.matrix([1,1,0])
math.intersect(a,b,c,d)
```
produces an error:
_Uncaught TypeError: Unexpected type of argument in function matrix (expected: string or Array or Matrix or number or boolean, actual: null, index: 0)_

However, when at least one of the parameters is an array or the lines DO actually intersect, intersect works:
```javascript
math.intersect(a.toArray(), b, c, d)
```
yields null.
```javascript
math.intersect(a, b, c, math.matrix([2,5,6]))
```
also correctly returns [1,3,3] (as a matrix).

Tried with math.js version 9.4.3.

**Comment by m93a** [comment]
Hey Robert, thanks for catching this! I'll look into the bug and see what I can do :)

**Comment by trent2** [comment]
Hey Michal, thank you for the very quick response and the fix! :+1:

**Comment by josdejong** [comment]
Fixed of @m93a is now published in `v9.4.4`
